### PR TITLE
Added navigation

### PR DESCRIPTION
**Pull request by fedeGiussa25**
## Summary

Added Navigation in the App

#### Galaxy S (or similar)
Peek 2020-07-20 16-51

#### Pixel 2 (or similar)
Peek 2020-07-20 16-38

#### Pixel 3 or 3XL (or similar)
Peek 2020-07-20 16-59

### iOS
- Not for this training.

## Trello Card

https://trello.com/c/zPYLweZB/13-n-31-introducci%C3%B3n-a-react-navigation

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -1,7 +1,13 @@
 import React from 'react';
+import 'react-native-gesture-handler';
```
Comment (mattgle) [inline App.tsx:2]: this should be at the line 1.

File: App.tsx [hunk 2]
```diff
@@ -1,7 +1,13 @@
...
+import { NavigationContainer } from '@react-navigation/native';
 
 import App from './src/app';
 
 export default function index() {
-  return <App />;
+  return (
+    <NavigationContainer>
+      <App />
+    </NavigationContainer>
```
Comment (mattgle) [inline App.tsx:11]: i think the NavigationContainer should be at the file ` src/app/index.tsx `

File: src/app/components/BookList/components/Book/index.tsx [hunk 3]
```diff
@@ -1,23 +1,26 @@
...
 
 import styles from './styles';
 
 const Book = ({ title, author, image }: BookProps) => {
   const sourceImage = image ? { uri: image } : imagePlaceholder;
+  const navigation = useNavigation();
 
   return (
-    <View style={styles.container}>
+    <TouchableOpacity onPress={() => navigation.navigate(ROUTES.DETAIL)} style={styles.container}>
```
Comment (mattgle) [inline src/app/components/BookList/components/Book/index.tsx:15]: extract the arrow function

File: src/app/index.tsx [hunk 4]
```diff
@@ -1,17 +1,22 @@
 import React from 'react';
-import { View, StatusBar } from 'react-native';
-import BookList from '@app/components/BookList';
-import { BOOK_LIST } from '@constants/books';
+import { StatusBar } from 'react-native';
+import { createStackNavigator } from '@react-navigation/stack';
+import Home from '@app/screens/Home';
+import Detail from '@app/screens/Detail';
```
Comment (mattgle) [inline src/app/index.tsx:5]: i think you can change the` '@app/screens/...'` to `'@screens/...'`

File: src/app/index.tsx [hunk 5]
```diff
@@ -1,17 +1,22 @@
...
     <>
       <StatusBar barStyle="dark-content" />
-      <View style={styles.container}>
-        <BookList data={BOOK_LIST} />
-      </View>
+      <Stack.Navigator initialRouteName={ROUTES.HOME} screenOptions={{ cardStyle: styles.container }}>
+        <Stack.Screen name={ROUTES.HOME} component={Home} />
+        <Stack.Screen name={ROUTES.DETAIL} component={Detail} />
+      </Stack.Navigator>
     </>
```
Comment (mattgle) [inline src/app/index.tsx:20]: i think the NavigationContainer should be here.

File: src/app/components/BookList/components/Book/index.tsx [hunk 6]
```diff
@@ -1,23 +1,27 @@
...
 import { BookProps } from '@interfaces/books';
 import imagePlaceholder from '@assets/img_book6.png';
+import ROUTES from '@constants/routes';
 
 import styles from './styles';
 
 const Book = ({ title, author, image }: BookProps) => {
   const sourceImage = image ? { uri: image } : imagePlaceholder;
+  const navigation = useNavigation();
+  const navigate = () => navigation.navigate(ROUTES.DETAIL);
```
Comment (sfernandez11) [inline src/app/components/BookList/components/Book/index.tsx:13]: Use a more descriptive name, for example, `handleGoToDetail` or `handleNavigateToDetail` or something like that

File: src/app/index.tsx [hunk 7]
```diff
@@ -1,17 +1,26 @@
...
-        <BookList data={BOOK_LIST} />
-      </View>
+      <NavigationContainer>
+        <Stack.Navigator initialRouteName={ROUTES.HOME} screenOptions={{ cardStyle: styles.container }}>
+          <Stack.Screen name={ROUTES.HOME} component={Home} />
+          <Stack.Screen name={ROUTES.DETAIL} component={Detail} />
+        </Stack.Navigator>
+      </NavigationContainer>
     </>
   );
```
Comment (sfernandez11) [inline src/app/index.tsx:25]:
```
  return (
    <NavigationContainer> 
      <StatusBar barStyle="dark-content" />
      <Stack.Navigator initialRouteName={ROUTES.HOME} screenOptions={{ cardStyle: styles.container }}>
        <Stack.Screen name={ROUTES.HOME} component={Home} />
        <Stack.Screen name={ROUTES.DETAIL} component={Detail} />
      </Stack.Navigator>
    </NavigationContainer>
  );
```

File: src/app/screens/Detail/index.tsx [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { Text } from 'react-native';
+
+const Detail = () => {
```
Comment (sfernandez11) [inline src/app/screens/Detail/index.tsx:4]: For screens we use `function` not arrow function

File: src/app/screens/Home/index.tsx [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import BookList from '@app/components/BookList';
+import { BOOK_LIST } from '@constants/books';
+
+const Home = () => {
```
Comment (sfernandez11) [inline src/app/screens/Home/index.tsx:5]: Same

File: src/app/index.tsx [hunk 10]
```diff
@@ -1,17 +1,26 @@
...
+import Detail from '@app/screens/Detail';
+import ROUTES from '@constants/routes';
 
 import styles from './styles';
 
+const Stack = createStackNavigator();
+
 const App = () => {
   return (
     <>
```
Comment (sfernandez11) [inline src/app/index.tsx:16]: The Fragment is not necessary

**Pull request merged**